pkg/codegen/pcl/invoke.go

```diff
@@ -44,6 +44,56 @@ func getInvokeToken(call *hclsyntax.FunctionCallExpr) (string, hcl.Range, bool)
 	return literal.Val.AsString(), call.Args[0].Range(), true
 }
 
+// annotateObjectProperties annotates the properties of an object expression with the types of the corresponding
+// properties in the schema. This is used to provide type information
+// for invoke calls that didn't have type annotations.
+//
+// this function will recursively annotate the properties of objects
+// that are nested within the object expression type.
+func annotateObjectProperties(modelType model.Type, schemaType schema.Type, topLevelAnnotate bool) {
```

Comment:
I don't entirely understand what `topLevelAnnotate` does. It looks like it is possible for multiple objects in a tree to be annotated. Is it a way to skip annotating the first `model.Type` encountered iff that is an `model.ObjectType`?

Reply:
Indeed the idea was to skip the top-level object but now I've removed this `topLevelAnnotate` parameter entirely and refactored the code with switch statements ✅